doc/config.rst

```diff
@@ -1753,6 +1754,15 @@ These options influence LaTeX output. See further :doc:`latex`.
 
    .. versionadded:: 1.6
 
+.. confval:: latex_use_wrapfig
+
+   The default is ``True``: figures with ``:align:`` option specifying to use
+   left or right alignment will be rendered using ``wrapfig`` macros. But,
+   this is known to have layouting issues in some cases. Set the option to
+   ``False`` to avoid usage of ``wrapfig``.
```

Comment:
clearly it is bad to have only a global setting. The choice should be made per figure, default being to use wrapfig and only in case of detected problems to not use it.

But it is impossible for Sphinx to detect problem, only to detect potential problem: if the figure follows immediately or precedes a list in particular.

Reply:
I think reST is a kind of lightweight mark up. Mainly it is marked up text. So I'd like not to add layout notations and directive options to reST.
Anyway this global option would help who has layouting problem. so +1 for this.

Reply:
I propose to add a LaTeX boolean which can be inserted by user to fix only one figure

```rst
.. raw:: latex

   \sphinxusewrapfigfalse

.. figure:: foo.png
   :align: right

.. raw:: latex

   \sphinxusewrapfigtrue
```

what do you think? I know it requires user to realize PDF had layout problem and then do the manual intervention.

Reply:
ah sorry, I need to think more. This method needs sphinx to provide mother environment which expands either to figure or wrapfigure.

Reply:
I feel using `raw` decreases the readability and writability.
So I choose the directive option rather than it.

Anyway, this discussion is not related with this PR directly. So let's file a new issue.